doc/specs/stdlib_hash_maps.md

```diff
@@ -0,0 +1,3088 @@
+is an `intent(out)` argument.
+
+##### Example
+
+```fortran
+    program demo_free_key
+      use stdlib_32_bit_key_data_wrapper, only: &
+          copy_key, free_key, key_type, set
+      use iso_fortran_env, only: int8
+      implicit none
+      integer(int8), allocatable :: value(:)
+      type(key_type) :: key_in, key_out
+      integer(int_8) :: i
+      allocate( value(1:15) )
+      do i=1, 15
+        value(i) = i
+      end do
+      call set( key_in, value )
+      call copy_key( key_in, key_out )
+      call free_key( key_out )
+    end program demo_free_key
+```
+
+#### `free_other` - frees the memory associated with other data
+
+##### Status
+
+Experimental
+
+##### Description
+
+Deallocates the memory associated with an variable of type
+`other_type`.
+
+##### Syntax
+
+`call [[stdlib_32_bit_key_data_wrapper:free_other]]( other )`
+
+##### Class
+
+Subroutine.
+
+##### Argument
+
+`other`: shall be a scalar variable of type `other_type`. It
+is an `intent(out)` argument.
+
+##### Example
+
+```fortran
+    program demo_free_other
+      use stdlib_32_bit_key_data_wrapper, only: &
+          copy_other, free_other, other_type, set
+      use iso_fortran_env, only: int8
+      implicit none
+      integer(int8), allocatable :: value(:)
+      type(key_type) :: other_in, other_out
+      integer(int_8) :: i
+      allocate( value(1:15) )
+      do i=1, 15
+          value(i) = i
+      end do
+      call set( other_in, value )
+      call copy_other( other_in, other_out )
+      call free_other( other_out )
+    end program demo_free_other
+```
+
+
+#### `get` - extracts the data from a derived type
+
+##### Status
+
+Experimental
+
+##### Description
+
+Extracts the data from a `key_type` or an `other_type` and stores it
+in the variable `value`..
+
+##### Syntax
+
+`call [[stdlib_32_bit_key_data_wrapper:get]]( key, value )`
+
+or
+
+`call [[stdlib_32_bit_key_data_wrapper:get]]( other, value )`
+
+
+##### Class
+
+Subroutine.
+
+##### Argument
+
+`key`: shall be a scalar expression of type `key_type`. It
+is an `intent(in)` argument.
+
+`other`: shall be a scalar expression of type `other_type`. It
+is an `intent(in)` argument.
+
+`value`: shall be an allocatable default character string variable, or
+an allocatable vector variable of type integer and kind `INT8`. It is
+an `intent(out)` argument.
+
+##### Example
+
+```fortran
+    program demo_get
+      use stdlib_32_bit_key_data_wrapper, only: &
+          get, key_type, set
+      use iso_fortran_env, only: int8
+      implicit none
+      integer(int8), allocatable :: value(:), result(:)
+      type(key_type) :: key
+      integer(int_8) :: i
+      allocate( value(1:15) )
+      do i=1, 15
+        value(i) = i
+      end do
+      call set( key, value )
+      call get( key, result )
+      print *, `RESULT == VALUE = ', all( value  == result )
+    end program demo_get
+```
+
+
+#### `HASHER_FUN`- serves aa function prototype.
```

Comment:
typo

Reply:
Fixed several typos in the documentation for `hasher_fun`.